### ISSUE TITLE
FreeTDS & TinyTDS

### ISSUE THREAD

**Issue by gengle** [rails-sqlserver/tiny_tds#511]
There has been many changes with SQL Server lately with the System.Data.SqlClient rewrite.  I recommend that we replace the FreeTDS, TinyTDS and design a new facade using .NET running on windows and linux.  marshalling between .NET & Ruby won't be too difficult - we can use uds, sockets, or grpc, etc.

Let me know if you'd like help with any of this.

**Comment by wpolicarpo** [comment]
What are the of what you are proposing over FreeTDS? Also, wouldn't it make more sense to write a new binding for that implementation rather than rewriting TinyTDS?

**Comment by gengle** [comment]
I was thinking we’d eliminate TinyTDS entirely.  The codebase is dating itself, references ot synbase and all the low level protocol boilderpoint code worries me.  Instead, what I’d recommend is slowly swap FreeTDS out with a new middleware driver using C# - hopefully the contract between this C# driver and TinyTDS would be minimal.

Some background.  I stubbled on to your project as I was trying to find a working rails sample that I could get working with a graphql service.  There are plenty of ActiveRecord samples using other rdms but for some reason MSSQL isn’t very popular and the samples I did find had integration issues or was not implemented fully.

I’m a strong Java, c#, MSSQL developer… a bit green on the Ruby side but I’d like to help where I can.


Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows

**Comment by aharpervc** [comment]
I've speculated about this exact concept myself. I think it's interesting. But I agree with @wpolicarpo, it'd probably just need to be a new from-scratch project. I think that designing a new gem is a much better use of time than trying to retrofit a new driver backend into TinyTDS itself.

There's not much ruby code in here anyway even if you did want to preserve it. `client.rb` is only 136 lines and the majority is version mapping.

**Comment by gengle** [comment]
Exactly.  Sounds like a plan to me.  I’m tied up the next couple weeks but hopefully next month sometime I’ll be able to take a crack at this.  Do you have a preference on IPC used?  gprc is what I’d probably recommend.

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows

**Comment by brodjustice** [comment]
Did this idea progress any further?

**Comment by gengle** [comment]
@brodjustice apologies as with my schedule I wasn't able to focus any time on this.  I might consider exploring this again if there is an interest.

**Comment by aae42** [comment]
> @brodjustice apologies as with my schedule I wasn't able to focus any time on this. I might consider exploring this again if there is an interest.

consider this interest

if i were to solve the problem tiny_tds solves today, i would use your design... c# sql client is basically the gold standard, c#/ruby interface of grpc seems logical

the other option is using go sql client to compile c stuff to use with ffi gem, but the go sql clients aren't as nice

**Comment by Michoels** [comment]
This would be great, but it sounds like a massive undertaking.

**Comment by andyundso** [comment]
I am closing this issue. As expressed in this issue, this likely needs an entirely new gem, which is outside of the scope for this repository.
